Specifications/Applications/for_Application/when_comparing_applications_with_different_names.cs

```diff
@@ -0,0 +1,25 @@
+using Machine.Specifications;
+
+namespace Dolittle.Applications.Specs.for_Application
+{
+    public class when_comparing_applications_with_different_names
+    {
+        const string name_a = "ApplicationA";
+        const string name_b = "ApplicationB";
+        static IApplication application_a, application_b;
+
+        Establish context = () => 
+        {
+            application_a = Application.WithName(name_a)
+                .Build();
+            application_b = Application.WithName(name_b)
+                .Build();
+        };
+
+        It should_hold_name_a = () => application_a.Name.Value.ShouldEqual(name_a);
```

Comment:
The spec is doing too much.  It says it is about comparing applications with different names, but there is no "Because of" that performs any comparison.

There should be a spec that checks the building, make sure names are assigned properly and a spec that does equality / gethashcode.